src/Blazor.Extensions.SignalR/HubConnection.cs

```diff
@@ -37,21 +37,120 @@ public HubConnection(HttpConnectionOptions options)
         {
             if (string.IsNullOrEmpty(methodName)) throw new ArgumentNullException(nameof(methodName));
             if (handler == null) throw new ArgumentNullException(nameof(handler));
 
             var callbackId = Guid.NewGuid().ToString();
 
             var callback = new HubMethodCallback(callbackId, methodName, this,
-                 (json) =>
+                 (payloads) =>
                  {
-                     var payload = Json.Deserialize<TResult>(json);
-                     return handler(payload);
+                     TResult1 t1 = default;
+                     TResult2 t2 = default;
+                     TResult3 t3 = default;
+                     TResult4 t4 = default;
+                     TResult5 t5 = default;
+                     TResult6 t6 = default;
+                     TResult7 t7 = default;
+                     TResult8 t8 = default;
+                     TResult9 t9 = default;
+                     TResult10 t10 = default;
+
+                     if (payloads.Length > 0)
+                     {
+                         t1 = Json.Deserialize<TResult1>(payloads[0]);
+                     }
+                     if (payloads.Length > 1)
+                     {
+                         t2 = Json.Deserialize<TResult2>(payloads[1]);
+                     }
+                     if (payloads.Length > 2)
+                     {
+                         t3 = Json.Deserialize<TResult3>(payloads[2]);
+                     }
+                     if (payloads.Length > 3)
+                     {
+                         t4 = Json.Deserialize<TResult4>(payloads[3]);
+                     }
+                     if (payloads.Length > 4)
+                     {
+                         t5 = Json.Deserialize<TResult5>(payloads[4]);
+                     }
+                     if (payloads.Length > 5)
+                     {
+                         t6 = Json.Deserialize<TResult6>(payloads[5]);
+                     }
+                     if (payloads.Length > 6)
+                     {
+                         t7 = Json.Deserialize<TResult7>(payloads[6]);
+                     }
+                     if (payloads.Length > 7)
+                     {
+                         t8 = Json.Deserialize<TResult8>(payloads[7]);
+                     }
+                     if (payloads.Length > 8)
+                     {
+                         t9 = Json.Deserialize<TResult9>(payloads[8]);
+                     }
+                     if (payloads.Length > 9)
+                     {
+                         t10 = Json.Deserialize<TResult10>(payloads[9]);
+                     }
+
+                     return handler(t1, t2, t3, t4, t5, t6, t7, t8, t9, t10);
                  }
             );
 
+            RegisterHandle(methodName, callback);
+
+            return callback;
+        }
+
+        internal void RegisterHandle(string methodName, HubMethodCallback callback)
```

Comment:
NIT: That could become an inline function as it is only used on the previous method. Will look at that later on.